### PR TITLE
Home_task_lottery_try_#1

### PR DESCRIPTION
**Pull request by slava-moiseev-kramatorsk**

### CODE REVIEW

File: src/main/java/core/basesyntax/Ball.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Ball {
+
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Ball.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        int index = new Random().nextInt(Colors.values().length);
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: move Random to the class field

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
+        return Colors.values()[index].toString();
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ColorSupplier.java:8]: see checklist
use name()

File: src/main/java/core/basesyntax/Colors.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+public enum Colors {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Colors.java:3]:
```suggestion
public enum Color {
```
all class name in singular

File: src/main/java/core/basesyntax/Lottery.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Lottery.java:6]:
```suggestion
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    public Ball getRandomBall() {
+        return new Ball(new Random().nextInt(100), new ColorSupplier().getRandomColor().toString());
+    }
+
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Lottery.java:10]:
```suggestion
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+}
+
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Lottery.java:12]:
```suggestion
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+
+    public Ball getRandomBall() {
+        return new Ball(new Random().nextInt(100), new ColorSupplier().getRandomColor().toString());
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Lottery.java:8]:
```suggestion
        return new Ball(new Random().nextInt(100), new ColorSupplier().getRandomColor());
```

no need again getting string. You are return string from method getRandomColor()

Comment (kozhukhovsky) [reply src/main/java/core/basesyntax/Lottery.java:8]: move random and colorsupplier to the class field

Comment (kozhukhovsky) [reply src/main/java/core/basesyntax/Lottery.java:8]: make 100 as class constant

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]
https://github.com/mate-academy/jv-lottery/blob/master/checklist.md

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 9]
```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
+    private final Random random = new Random();
+    private final int index = random.nextInt(Color.values().length);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: You need to generate a new index for each method invocation - you currently generate it only once, when the class is created, and then getRandomColor returns that color each time the method is called. Make it a local variable instead

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job so far!

File: src/main/java/core/basesyntax/Lottery.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private final int maxBalls = 100;
```
[the remark this answers is not in the document]
Comment (slava-moiseev-kramatorsk) [reply src/main/java/core/basesyntax/Lottery.java:6]: image

[the remark this answers is not in the document]
Comment (slava-moiseev-kramatorsk) [reply src/main/java/core/basesyntax/Lottery.java:6]: Intellij recommends moving it to a method

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**